learn/guides/organizing-ballerina-code.md

```diff
@@ -0,0 +1,144 @@
+
+`bal new` command creates the Package with the default module. This will generate the `Ballerina.toml` which identifies a directory as a package and will additionally generate a sample source file in the default module.
+
+### Creating your first Ballerina package
+
+You can create a Ballerina package with the `bal new` command as follows. 
+
+```bash
+bal new hello_world
+```
+
+This will create a new Ballerina package with a main function. 
+
+The `bal new` command generates the following file structure.
+
+```bash
+cd hello_world
+tree .
+.
+├── Ballerina.toml
+└── main.bal
+    
+0 directories, 2 files
+```
+
+You may also try creating a service or a lib package instead of a main function. 
+
+#### Creating a Ballerina service package
+
+```bash
+bal new -t service hello_service
+```
+
+This will create a Ballerina source containing a service declaration with Ballerina tests to test the service. The following file structure will be generated with the service template.  
+
+```bash
+cd hello_service
+.
+├── Ballerina.toml
+├── service.bal
+└── tests
+    └── service_test.bal
+
+1 directory, 3 files
+```
+
+#### Creating a library Package
+
+```bash
+bal new -t lib hello_lib
+```
+
+This will create a Ballerina source file containing a function that prints `Hello, world!` along with a test file to test the function. 
+Additionally, it will also create the `Package.md` file which is required to [publish a package to Ballerina Central]((/learn/user-guide/publishing-packages-to-ballerina-central).
+
+```bash
+
+├── Ballerina.toml
+├── hello_lib.bal
+├── Module.md
+├── Package.md
+├── resources
+└── tests
+    └── lib_test.bal
+```
+
+### The Default Module
+
+When a package is created with `bal new` command, the `Ballerina.toml` and the `main.bal` is created. The `main.bal` file is a Ballerina source file which belongs to the default module. The root directory of the default module is the root directory of the package as well. 
+Therefore, the package root directory contains files that belong to the package as well the default module.
+
+You can add more source files at the root and symbols such as functions and variables in one file are visible to other files because they are in the same namespace. This namespace is called the default module of the package. The package name, which is specified in the Ballerina.toml file, is also used to refer to the default module.
+
+### Non-default modules
+
+As projects grow in complexity, the need arises to organize code better. This could be because you want to separate the functionalities of the package and/or to add boundaries to visibility of certain functionalities. Therefore, Ballerina allows subdividing the code into multiple modules as well.
```

Comment:
```suggestion
As projects grow in complexity, the need arises to organize code better. This could be because you want to separate the functionalities of the package and/or to add boundaries to the visibility of certain functionalities. Therefore, Ballerina allows subdividing the code into multiple modules as well.
```